src/neuroconv/utils/json_schema.py

```diff
@@ -70,104 +69,93 @@ def get_base_schema(
     return base_schema
 
 
-def get_schema_from_method_signature(method: Callable, exclude: list = None) -> dict:
+def get_schema_from_method_signature(method: Callable, exclude: list[str] | None = None) -> dict:
+    """Deprecated version of `get_json_schema_from_method_signature`."""
+    message = (
+        "The method `get_schema_from_method_signature` is now named `get_json_schema_from_method_signature`."
+        "This method is deprecated and will be removed after January 2025."
+    )
+    warnings.warn(message=message, category=DeprecationWarning, stacklevel=2)
+
+    return get_json_schema_from_method_signature(method=method, exclude=exclude)
+
+
+def get_json_schema_from_method_signature(method: Callable, exclude: list[str] | None = None) -> dict:
     """
-    Take a class method and return a json-schema of the input args.
+    Get the equivalent JSON schema for a signature of a method.
+
+    Also uses `docstring_parser` (NumPy style) to attempt to find descriptions for the arguments.
 
     Parameters
     ----------
-    method: function
-    exclude: list, optional
+    method : callable
+        The method to generate the JSON schema from.
+    exclude : list of str, optional
+        List of arguments to exclude from the schema generation.
+        Always includes 'self' and 'kwargs'.
 
     Returns
     -------
-    dict
-
+    json_schema : dict
+        The JSON schema corresponding to the method signature.
     """
-    if exclude is None:
-        exclude = ["self", "kwargs"]
-    else:
-        exclude = exclude + ["self", "kwargs"]
-    input_schema = get_base_schema()
-    annotation_json_type_map = dict(
-        bool="boolean",
-        str="string",
-        int="number",
-        float="number",
-        dict="object",
-        list="array",
-        tuple="array",
-        FilePathType="string",
-        FolderPathType="string",
+    exclude = exclude or []
+    exclude += ["self", "cls"]
+
+    signature = inspect.signature(obj=method)
+    parameters = signature.parameters
+    additional_properties = False
+    arguments_to_annotations = {}
+    for argument_name in parameters:
+        if argument_name in exclude:
+            continue
+
+        parameter = parameters[argument_name]
+
+        if parameter.kind == inspect.Parameter.VAR_KEYWORD:  # Skip all **{...} usage
+            additional_properties = True
+            continue
+
+        annotation = parameter.annotation
+
+        # Pydantic uses ellipsis for required
+        pydantic_default = ... if parameter.default is inspect._empty else parameter.default
+
+        arguments_to_annotations.update({argument_name: (annotation, pydantic_default)})
+
+    model = pydantic.create_model(
+        "_TempModel", __config__=pydantic.ConfigDict(arbitrary_types_allowed=True), **arguments_to_annotations
```

Comment:
If you're wondering about the config here, it is a requirement to support any numpy types (which, how we do it here, could also be improved such as in https://github.com/pydantic/pydantic/issues/7017)

For example, https://github.com/catalystneuro/neuroconv/blob/main/src/neuroconv/datainterfaces/ecephys/spikegadgets/spikegadgetsdatainterface.py#L27

Reply:
Note we aren't using any Pydantic-based validation here (saving that for 3rd PR of this chain) but when we do add that - yes, those do get skipped - however, they are not skipped when generating a JSON schema for the class, which then fails if it can't figure out an appropriate type to associate

Reply:
Yep. This is what I failed to realize. We still need json.  

Also, the suggestion disapleyd wrong. What I was suggesting was to pass the model_name as keyword: `model_name="_TempModel"` instead of positionally as we are doing here.

Reply:
@h-mayorquin You know I would love nothing more than that, and it's a rare day when you catch me doing anything but

However, as I mentioned elsewhere, Pydantic loves their first positional arguments 😂 So it is actually not allowed at all in this case, much to my dismay